tests/test_utils.py

```diff
@@ -35,3 +35,24 @@ def test_given_args_expected_result(self, input_str: str):
 
     def test_given_valid_args_returns_true(self):
         assert str_is_empty_or_none("aValue") is False
+
+
+class TestGetListIndexSafely:
+    @pytest.mark.parametrize(
+        "list, item, start, end, exp_index",
+        [
+            pytest.param([0, 1, 2], 0, 0, 2, 0),
+            pytest.param([0, 1, 2], 1, 0, 2, 1),
+            pytest.param([0, 1, 2], 2, 0, 2, 2),
+            pytest.param([0, 1, 2], 3, 0, 2, -1),
+            pytest.param([0, 1, 2], 0, 0, 1, 0),
+            pytest.param([0, 1, 2], 1, 0, 1, 1),
```

Comment:
If you'd change the behavior of the ```end``` argument (see other review remark), then this test should return -1.

Reply:
I have removed this function because it was not needed anymore with the parser changes.